### ISSUE TITLE
Basic auth middleware error with Django 1.11 - ContentNotRenderedError

### ISSUE THREAD

**Issue by tm-kn** [aldryn/aldryn-sso#50]
I tested with Aldryn SSO 1.1.16 with the following settings:

```python
ALDRYN_SSO_OVERIDE_ADMIN_LOGIN_VIEW = False
ALDRYN_SSO_ALWAYS_REQUIRE_LOGIN = 'basicauth'
ALDRYN_SSO_BASICAUTH_USER = 'user'
ALDRYN_SSO_BASICAUTH_PASSWORD = 'pass'
```

it seems like it may be fixed by rendering the response in there by calling `render()` on `TemplateResponse`.

https://github.com/aldryn/aldryn-sso/blob/01105801fa6ee7c6f7d3491d7bc10ab7e8e1c89d/aldryn_sso/middleware.py#L138-L147

Error
```
Environment:


Request Method: GET
Request URL: https://xxxxxxxxxxxx-stage.us.aldryn.io/admin

Django Version: 1.11.10
Python Version: 3.6.3
Installed Applications:
['aldryn_addons',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.sites',
 'django.contrib.messages',
 'aldryn_sso',
 'django.contrib.admin',
 'django.contrib.staticfiles',
 'aldryn_django',
 'aldryn_sites',
 'raven.contrib.django',
 'wagtail.wagtailforms',
 'wagtail.wagtailredirects',
 'wagtail.wagtailembeds',
 'wagtail.wagtailsites',
 'wagtail.wagtailusers',
 'wagtail.wagtailsnippets',
 'wagtail.wagtaildocs',
 'wagtail.wagtailimages',
 'wagtail.wagtailsearch',
 'wagtail.wagtailadmin',
 'wagtail.wagtailcore',
 'modelcluster',
 'taggit',
 'xxxxxxxxxxxx.authentication',
 'xxxxxxxxxxxx.blockpages',
 'xxxxxxxxxxxx.categories',
 'xxxxxxxxxxxx.content',
 'xxxxxxxxxxxx.esi',
 'xxxxxxxxxxxx.documents',
 'xxxxxxxxxxxx.forms',
 'xxxxxxxxxxxx.home',
 'xxxxxxxxxxxx.images',
 'xxxxxxxxxxxx.navigation',
 'xxxxxxxxxxxx.search',
 'xxxxxxxxxxxx.units',
 'xxxxxxxxxxxx.utils',
 'xxxxxxxxxxxx.years',
 'wagtail.contrib.modeladmin',
 'wagtail.contrib.settings',
 'wagtail.contrib.wagtailsearchpromotions',
 'captcha',
 'wagtailcaptcha']
Installed Middleware:
['raven.contrib.django.middleware.DjangoRestFrameworkCompatMiddleware',
 'raven.contrib.django.middleware.SentryMiddleware',
 'django.middleware.gzip.GZipMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'aldryn_sso.middleware.BasicAuthAccessControlMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'aldryn_django.middleware.LanguagePrefixFallbackMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.contrib.sites.middleware.CurrentSiteMiddleware',
 'aldryn_sites.middleware.SiteMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'wagtail.wagtailcore.middleware.SiteMiddleware',
 'wagtail.wagtailredirects.middleware.RedirectMiddleware']



Traceback:

File "/virtualenv/lib/python3.6/site-packages/django/core/handlers/base.py" in get_response
  131.                 response = middleware_method(request, response)

File "/virtualenv/lib/python3.6/site-packages/django/middleware/common.py" in process_response
  139.             response['Content-Length'] = str(len(response.content))

File "/virtualenv/lib/python3.6/site-packages/django/template/response.py" in content
  129.                 'The response content must be rendered before it can be accessed.'

Exception Type: ContentNotRenderedError at /admin
Exception Value: The response content must be rendered before it can be accessed.
```

Failing in - https://github.com/django/django/blob/01448a97e075df08ed0f89c443ee35e649dfb630/django/template/response.py#L127-L131

**Comment by evildmp** [comment]
@tm-kn " tested with 1.1.16" - do you mean "1.11.6"?

**Comment by filwaitman** [comment]
Hey @tm-kn, thanks for the report!

This is fixed on aldryn-sso 1.2.1 (beta).
